internal/compiler/widgets/fluent/std-widgets.slint

```diff
@@ -354,12 +356,43 @@ export TabImpl := Rectangle {
         x: (parent.width - width) / 2;
     }
 
-    fs := FocusScope {}
+    Rectangle { // Focus Rectangle
+        border-width: enabled && has-focus ? 1px : 0px;
+        border-color: Palette.black;
+    }
 }
 
-export TabBarImpl := HorizontalLayout {
-    spacing: 8px;
-    alignment: start;
+export TabBarImpl := Rectangle {
+    // injected properties:
+    property<int> current; // The currently selected tab
+    property<int> current-focused: fs.has-focus ? fs.focused-tab : -1; // The currently focused tab
+    property<int> num-tabs; // The total number of tabs
+
+    HorizontalLayout {
+        spacing: 8px;
+        alignment: start;
+        @children
+    }
+
+    fs := FocusScope {
+        property<int> focused-tab: 0;
+
+        key-pressed(event) => {
+            if (event.text == " " || event.text == "\n") {
```

Comment:
I think activation should happen on key release, not press. Just like a button is "clicked" on release, provided that it was pressed before (via mouse or key press).

Reply:
The newest version is closer. Return triggers on press (so do the cursor keys), space on release.

Missing: The "pressed" state on the tab while space is pressed.